bigtable-hbase-2.x-parent/bigtable-hbase-2.x-integration-tests/src/test/java/com/google/cloud/bigtable/hbase/TestBatch.java

```diff
@@ -15,10 +15,28 @@
  */
 package com.google.cloud.bigtable.hbase;
 
+import static org.junit.Assert.assertNull;
+
+import com.google.common.collect.ImmutableList;
 import org.apache.hadoop.hbase.client.Append;
+import org.apache.hadoop.hbase.client.Table;
+import org.junit.Test;
 
 public class TestBatch extends AbstractTestBatch {
   protected void appendAdd(Append append, byte[] columnFamily, byte[] qualifier, byte[] value) {
     append.addColumn(columnFamily, qualifier, value);
   }
+
+  @Test
+  public void testBatchWithMismatchedResultArray() throws Exception {
+    Table table = getDefaultTable();
+    Exception actualError = null;
+    try {
+      // This is accepted behaviour in HBase 2 API, It ignores the `new Object[1]` param.
+      table.batchCallback(ImmutableList.of(), new Object[1], null);
```

Comment:
I think you should verify the interaction with a callback

Reply:
Do you mean something like this:
```java
@Test
  public void testBatchWithMismatchedResultArray() throws Exception {
    Table table = getDefaultTable();
    AtomicBoolean isCallbackInvoked = new AtomicBoolean(false);
    SettableFuture<Void> future = SettableFuture.create();
    Exception actualError = null;
    try {
      // This is accepted behaviour in HBase 2 API, It ignores the `new Object[1]` param.
      table.batchCallback(
          ImmutableList.of(),
          new Object[1],
          new Batch.Callback<Result>() {
            @Override
            public void update(byte[] region, byte[] row, Result result) {
              isCallbackInvoked.set(true);
              future.set(null);
            }
          });
      future.get(20, TimeUnit.SECONDS);
    } catch (Exception ex) {
      actualError = ex;
    }
    assertNull(actualError);
  }
```
I think it would be hard to verify the callback in this scenario. As HBase-2x's minicluster completely ignores the callback as List is empty. 

Or do you mean another test where we could verify the callback?

Reply:
I was thinking something like:

```java
Batch.Callback callback = Mockito.mock(Batch.Callback.class);
table.batchCallback(..., callback);
Mockito.verify(callback, Mockito.never()).update(Mockito.any(), Mockito.any(), Mockito.any());
```

Reply:
Thanks for the explanation, I updated test with the same.